src/Azure.Functions.Cli/Helpers/VersionHelper.cs

```diff
@@ -0,0 +1,141 @@
+                foreach (var item in releases)
+                {
+                    var jProperty = (Newtonsoft.Json.Linq.JProperty)item;
+                    var releaseDetail = JsonConvert.DeserializeObject<ReleaseDetail>(jProperty.Value.ToString());
+                    releaseList.Add(new ReleaseSummary() { Release = jProperty.Name, ReleaseDetail = releaseDetail.ReleaseList.FirstOrDefault() });
+                }
+
+                var currentCoreToolsRelease = releaseList.FirstOrDefault(x => x.CoreToolsReleaseNumber == Constants.CliDetailedVersion[..Constants.CliDetailedVersion.IndexOf(" ")]);
+                var latestCoreToolsRelease = releaseList.FirstOrDefault(x => x.Release == data.Tags.V4Release.ReleaseVersion);
+
+                if ( !string.IsNullOrEmpty(currentCoreToolsRelease?.CoreToolsReleaseNumber) && 
+                    !string.IsNullOrEmpty(latestCoreToolsRelease?.CoreToolsReleaseNumber) && 
+                    currentCoreToolsRelease.CoreToolsReleaseNumber != latestCoreToolsRelease.CoreToolsReleaseNumber)
+                {
+                    return Constants.OldCoreToolsVersionMessage;
+                }
+
+                return string.Empty;
+            }
+            catch (Exception)
+            {
+                return string.Empty;
+            }
+        }
+
+        private class CliFeed
+        {
+            [JsonProperty("tags")]
+            public Tags Tags { get; set; }
+
+            [JsonProperty("releases")]
+            public Object Releases { get; set; }
+        }
+
+        private class Tags
+        {
+            [JsonProperty("v4")]
+            public Release V4Release { get; set; }
+
+            [JsonProperty("v4-prerelease")]
+            public Release V4PreRelease { get; set; }
+
+            [JsonProperty("v3")]
+            public Release V3Release { get; set; }
+
+            [JsonProperty("v3-prerelease")]
+            public Release V3PreRelease { get; set; }
+        }
+
+        private class Release
+        {
+            [JsonProperty("release")]
+            public string ReleaseVersion { get; set; }
+
+            [JsonProperty("releaseQuality")]
+            public string ReleaseQuality { get; set; }
+
+            [JsonProperty("hidden")]
+            public bool Hidden { get; set; }
+        }
+
+        private class ReleaseSummary
+        {
+            public string Release { get; set; }
+
+            public string CoreToolsReleaseNumber
+            {
+                get
+                {
+                    var downloadLink = ReleaseDetail?.DownloadLink;
+                    if (string.IsNullOrEmpty(ReleaseDetail?.DownloadLink))
+                    {
+                        return string.Empty;
+                    }
+
+                    downloadLink = downloadLink.Replace("https://functionscdn.azureedge.net/public/", string.Empty);
+                    return downloadLink[..downloadLink.IndexOf('/')];
```

Comment:
Could you explain why we are truncating the base URL from the download link?

Reply:
The official core tools version in only available in the download link in the json file and it can only be fetched from there. I have, however, simplified the logic.